Node E2E: Remove kubelet related flags

Random-Liu opened the pull request:
This PR:
1. **Add `KubeletConfig` in test context.** The configuration is dynamically retrieved from kubelet `configz/` endpoint after test node registers itself.
2. **Change tests to look at `KubeletConfig` instead of other kubelet related fields in test context.**
3. **Remove kubelet related flags from node e2e.** Add `kubelet-flags` to pass kubelet flags all together.

Why do we need this change?
1. This is required by **node soaking test and node conformance test**.
  * **For node soaking test.** We'll have one job start the test environment, and another job running the actual test again and again. It's hard to tell the test by flag how kubelet is configured.
  * **For node conformance test.** Kubelet is usually started by user. It's hard and troublesome to tell the test by flag how kubelet is configured.
2. **No need to add another flag in node e2e for each kubelet flag we want to set in the test.** Just directly add the kubelet flag into `kubelet-flags`.

In the future,
1. Kubelet start logic should be moved outside of the test. We should have standard kubelet launcher or standard cloud-init node setup script etc.
2. The test should validate kubelet configuration:
  * Whether the configuration is production ready?
  * Whether the configuration has conflicts?
  * Whether some configuration conflict with node e2e/conformance test is turned on (cloud provider etc.)? (Some configurations need to be turned off when running node e2e/conformance test)
  * ...

/cc @kubernetes/sig-node

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36919)
<!-- Reviewable:end -->

Random-Liu committed:
Remove dependency on kubelet related flags.

Random-Liu committed:
Remove kubelet related flags from node e2e. Add a single flag `kubelet-flags` to pass kubelet flags all together.

Random-Liu commented:
This is a node e2e test framework change. It is required for node conformance test and node soaking test.

/cc @saad-ali

Random-Liu committed:
Update bazel.

saad-ali commented:
> This is a node e2e test framework change. It is required for node conformance test and node soaking test.
> 
> /cc @saad-ali

Ack, don't break anything :)

timstclair commented on test/e2e_node/services/kubelet.go:
```diff
@@ -0,0 +1,174 @@
+		"--serialize-image-pulls", "false",
+		"--config", manifestPath,
+		"--file-check-frequency", "10s", // Check file frequently so tests won't wait too long
+		"--pod-cidr=10.180.0.0/24", // Assign a fixed CIDR to the node because there is no node controller.
```
nit: use consistent argument format: `"--pod-cidr", "10.180.0.0/24"`

Random-Liu replied:
Done.

timstclair reviewed (commented):
Just a couple nits.

Random-Liu commented on test/e2e_node/services/kubelet.go:
```diff
@@ -0,0 +1,174 @@
+package services
```
Done

Random-Liu commented on test/e2e_node/services/kubelet.go:
```diff
@@ -0,0 +1,174 @@
+	if flag.Parsed() {
+		return nil
+	}
+	// Notice that we assume all white spaces in flag string are spliter.
```
Done

Random-Liu commented:
@timstclair Addressed comments and fixed one problem.

`feature-gates` should still be passed as`TEST_ARGS` instead of `KUBELET_ARGS`, because "test master" also relies on it.
Kubelet will inherit feature-gates of the test framework by default, but it could also be override with `kubelet-flags`.

timstclair commented:
LGTM

Random-Liu commented:
Rebased. Apply LGTM again.
